src/test/java/org/jitsi/meet/test/base/FailureListener.java

```
@@ -0,0 +1,362 @@
+     * @param fileName the destination html file name.
+     */
+    private void saveHtmlSource(WebDriver driver, String fileName)
+    {
+        try
+        {
+            FileUtils.openOutputStream(
+                new File(outputHtmlSourceParentFolder, fileName))
+                .write(driver.getPageSource().getBytes());
+        }
+        catch(Throwable t)
+        {
+            t.printStackTrace();
+        }
+    }
+
+    /**
+     * Saves the log from meet. Normally when clicked it is saved in Downloads
+     * if we do not find it we skip it.
+     */
+    private void saveMeetDebugLog(
+        String fileNamePrefix, AbstractBaseTest testInstance)
+    {
+        saveMeetDebugLog(testInstance.participant1.getDriver(),
+            fileNamePrefix + "-meetlog-owner.json");
+
+        if(testInstance.participant2 != null)
+        {
+            saveMeetDebugLog(testInstance.participant2.getDriver(),
+                fileNamePrefix + "-meetlog-participant.json");
+        }
+
+        if(testInstance.participant3 != null)
+        {
+            saveMeetDebugLog(testInstance.participant3.getDriver(),
+                fileNamePrefix + "-meetlog-third.json");
+        }
+    }
+
+    /**
+     * Saves the log from meet. Normally when clicked it is saved in Downloads
+     * if we do not find it we skip it.
+     */
+    private void saveMeetDebugLog(WebDriver driver, String fileName)
+    {
+        try
+        {
+            Object log = ((JavascriptExecutor) driver)
+                .executeScript(
+                    "try{ "
+                        + "return JSON.stringify(APP.conference.getLogs(), null, '    ');"
+                        + "}catch (e) {}");
+
+            if(log == null)
+                return;
+
+            FileUtils.write(
+                new File(outputLogsParentFolder, fileName),
+                (String)log);
+        }
+        catch (Exception e)
+        {
+            //e.printStackTrace();
```

why is that exception swallowed ?

---

Fixed